endpointmanager/pkg/chplquerier/vendorquerier_integration_test.go

```diff
@@ -0,0 +1,241 @@
+// +build integration
+
+package chplquerier
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strings"
+	"testing"
+
+	th "github.com/onc-healthit/lantern-back-end/endpointmanager/pkg/testhelper"
+	"github.com/pkg/errors"
+	logtest "github.com/sirupsen/logrus/hooks/test"
+)
+
+func Test_persistVendor(t *testing.T) {
+	teardown, _ := th.IntegrationDBTestSetup(t, store.DB)
+	defer teardown(t, store.DB)
+
+	var err error
+
+	var ctx context.Context
+	var cancel context.CancelFunc
+
+	chplVend := testCHPLVendor1
+	vend := testVendor1
+
+	var ct int
+	ctStmt, err := store.DB.Prepare("SELECT COUNT(*) FROM vendors;")
+	th.Assert(t, err == nil, err)
+	defer ctStmt.Close()
+
+	// check that ended context when no element in store fails as expected
+	ctx, cancel = context.WithCancel(context.Background())
+	cancel()
+	err = persistVendor(ctx, store, &chplVend)
+	th.Assert(t, errors.Cause(err) == context.Canceled, "should have errored out with root cause that the context was canceled")
+
+	err = ctStmt.QueryRow().Scan(&ct)
+	th.Assert(t, err == nil, err)
+	th.Assert(t, ct == 0, "should not have stored data")
+
+	// reset context
+	ctx = context.Background()
+
+	// check that new item is stored
+	err = persistVendor(ctx, store, &chplVend)
+	th.Assert(t, err == nil, err)
+
+	err = ctStmt.QueryRow().Scan(&ct)
+	th.Assert(t, err == nil, err)
+	th.Assert(t, ct == 1, "did not store data as expected")
+	storedVend, err := store.GetVendorUsingName(ctx, "Epic Systems Corporation")
+	th.Assert(t, err == nil, err)
+	th.Assert(t, vend.Equal(storedVend), "1stored data does not equal expected store data")
+
+	// check that a change replaces the item
+	chplVend.Status = chplStatus{ID: 2, Status: "other"}
+	vend.Status = "other"
+	err = persistVendor(ctx, store, &chplVend)
+	th.Assert(t, err == nil, err)
+
+	err = ctStmt.QueryRow().Scan(&ct)
+	th.Assert(t, err == nil, err)
+	th.Assert(t, ct == 1, "did not store data as expected")
+	storedVend, err = store.GetVendorUsingName(ctx, "Epic Systems Corporation")
+	th.Assert(t, err == nil, err)
+	th.Assert(t, vend.Equal(storedVend), "2stored data does not equal expected store data")
```

Comment:
This error message and the one above it should probably be changed to something like `"initial stored data does not equal expected store data"` and `"updated stored data does not equal expected store data"` or something.